packages/web/src/components/gcds-link/gcds-link.tsx

```diff
@@ -0,0 +1,220 @@
+
+  /**
+   * The type specifies the media type of the linked document
+   */
+  @Prop() type: string | undefined;
+
+  /**
+   * Custom callback function on click event
+   */
+  @Prop() clickHandler: Function;
+
+  /**
+   * Set additional HTML attributes not available in component properties
+   */
+  @State() inheritedAttributes: Object = {};
+
+  /**
+   * Language of rendered component
+   */
+  @State() lang: string;
+
+  /**
+   * Events
+   */
+
+  /**
+   * Emitted when the link has focus.
+   */
+  @Event() gcdsFocus!: EventEmitter<void>;
+
+  /**
+   * Emitted when the link loses focus.
+   */
+  @Event() gcdsBlur!: EventEmitter<void>;
+
+  /*
+   * Observe lang attribute change
+   */
+  updateLang() {
+    const observer = new MutationObserver(mutations => {
+      if (mutations[0].oldValue != this.el.lang) {
+        this.lang = this.el.lang;
+      }
+    });
+    observer.observe(this.el, observerConfig);
+  }
+
+  componentWillLoad() {
+    // Validate attributes and set defaults
+    this.validateSize(this.size);
+
+    this.inheritedAttributes = inheritAttributes(this.el, this.shadowElement, [
+      'referrerpolicy',
+    ]);
+
+    // Define lang attribute
+    this.lang = assignLanguage(this.el);
+
+    this.updateLang();
+  }
+
+  render() {
+    const {
+      size,
+      lang,
+      display,
+      href,
+      rel,
+      target,
+      external,
+      download,
+      type,
+      inheritedAttributes,
+    } = this;
+
+    const Tag = 'a';
+    const attrs = {
+      display,
+      href,
+      rel,
+      target,
+      download,
+      type,
+    };
+
+    const isExternal = target === '_blank' || external;
+
+    return (
+      <Host>
+        <Tag
+          {...attrs}
+          onClick={e => this.handleClick(e)}
+          class={`link--${size}`}
+          ref={element => (this.shadowElement = element as HTMLElement)}
+          target={isExternal ? '_blank' : target}
+          rel={isExternal ? 'noopener' : rel}
+          {...inheritedAttributes}
+          part="link"
+        >
+          <slot name="left"></slot>
+
+          <slot></slot>
+
+          {target === '_blank' || external ? (
+            <gcds-icon
+              name="external-link"
+              label={i18n[lang].external}
+              margin-left="200"
+            />
+          ) : download ? (
+            <gcds-icon
+              name="download"
+              label={i18n[lang].download}
+              margin-left="100"
+            />
+          ) : href.toLowerCase().startsWith('mailto:') ? (
+            <gcds-icon
+              iconStyle="regular"
+              name="envelope"
+              label={i18n[lang].email}
+              margin-left="100"
+            />
+          ) : href.toLowerCase().startsWith('tel:') ? (
+            <gcds-icon
+              name="phone"
+              label={i18n[lang].phone}
+              margin-left="100"
+            />
+          ) : (
+            <slot name="right"></slot>
```

Comment:
We can remove the slot, this was an icon slot for the button component that is not needed.

Reply:
This is what I'm planning on doing to remove this extra slot, but wondering if we could make the code better.
```suggestion
          ) : (
            href.toLowerCase().startsWith('tel:') && (
              <gcds-icon
                name="phone"
                label={i18n[lang].phone}
                margin-left="100"
              />
            )
          )}
```

Reply:
If you could send me a sample that be be awesome @melaniebmn